src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1007,6 +1007,8 @@ - (void)_handleSubscriptionEstablished
 {
     os_unfair_lock_lock(&self->_lock);
 
+    MATTER_LOG_METRIC_END(kMetricMTRDeviceSubscriptionSetup, CHIP_NO_ERROR);
```

Comment:
So this will get called on internal re-subscribes from the ReadClient, so we can get a MATTER_LOG_METRIC_END here without a corresponding START.  Is that OK?  If not, consider conditioning this on whether `HadSubscriptionEstablishedOnce()` is true (see code a bit below this).